credentials/credentials.go

```diff
@@ -190,16 +196,16 @@ func NewTLS(c *tls.Config) TransportCredentials {
 }
 
 // NewClientTLSFromCert constructs a TLS from the input certificate for client.
-// serverNameOverwrite is for testing only. If set to a non empty string,
-// it will overwrite the virtual host name of authority (e.g. :authority header field) in requests.
-func NewClientTLSFromCert(cp *x509.CertPool, serverNameOverwrite string) TransportCredentials {
-	return NewTLS(&tls.Config{ServerName: serverNameOverwrite, RootCAs: cp})
+// serverNameOverride is for testing only. If set to a non empty string,
+// it will override the virtual host name of authority (e.g. :authority header field) in requests.
+func NewClientTLSFromCert(cp *x509.CertPool, serverNameOverride string) TransportCredentials {
```

Comment:
Not sure if I understand your comment correctly.

If you mean we don't need function `NewClientTLSFromCert()`, that's probably true. It's just a wrapper of `NewTLS()`. But it helps to cover the implementing details of TLS creds.

If you mean we don't need the newly added API `OverrideServerName()`. grpclb needs to override the server name without knowing the real type of the creds (it could be something other than TLS), the new API is necessary to accomplish this.